docs/book/src/reference/watching-resources/testdata/external-indexed-field/controller.go

```diff
@@ -0,0 +1,210 @@
+- It needs to be able to fully manage Deployments, as well as check their status.
+- It also needs to be able to get, list and watch ConfigMaps.
+All 3 of these are important, and you will see usages of each below.
+*/
+
+// +kubebuilder:rbac:groups=apps.tutorial.kubebuilder.io,resources=configdeployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps.tutorial.kubebuilder.io,resources=configdeployments/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
+
+/*
+`Reconcile` will be in charge of reconciling the state of ConfigDeployments.
+ConfigDeployments are used to manage Deployments whose pods are updated whenever the configMap that they use is updated.
+
+For that reason we need to add an annotation to the PodTemplate within the Deployment we create.
+This annotation will keep track of the latest version of the data within the referenced ConfigMap.
+Therefore when the version of the configMap is changed, the PodTemplate in the Deployment will change.
+This will cause a rolling upgrade of all Pods managed by the Deployment.
+
+Skip down to the `SetupWithManager` function to see how we ensure that `Reconcile` is called when the referenced `ConfigMaps` are updated.
+*/
+func (r *ConfigDeploymentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	/* */
+	_ = context.Background()
+	_ = r.Log.WithValues("configdeployment", req.NamespacedName)
+
+	var configDeployment appsv1.ConfigDeployment
+	if err := r.Get(ctx, req.NamespacedName, &configDeployment); err != nil {
+		log.Error(err, "unable to fetch ConfigDeployment")
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	// +kubebuilder:docs-gen:collapse=Begin the Reconcile
+
+	// your logic here
+
+	var configMapVersion string
+	if configDeployment.Spec.ConfigMap != "" {
+		configMapName := configDeployment.Spec.ConfigMap
+		foundConfigMap := &corev1.ConfigMap{}
+		err := r.Get(context.TODO(), types.NamespacedName{Name: configMapName, Namespace: configDeployment.Namespace}, foundConfigMap)
+		if err != nil {
+			// If a configMap name is provided, then it must exist
+			// You will likely want to create an Event for the user to understand why their reconcile is failing.
+			return ctrl.Result{}, err
+		}
+
+		// Hash the data in some way, or just use the version of the Object
+		configMapVersion = foundConfigMap.ResourceVersion
+	}
+
+	// Logic here to add the configMapVersion as an annotation on your Deployment Pods.
+
+	/* */
+
+	return ctrl.Result{}, nil
+}
+// +kubebuilder:docs-gen:collapse=Finish the Reconcile
```

Comment:
I don't think hidding this part makes that much sense as it is just one LoC. Also, the function seems a bit odd without the closing curly bracket.